### PR TITLE
Fix some races in getNetworkFromStore

### PR DESCRIPTION
**Pull request by LK4D4**

### CODE REVIEW

File: endpoint.go [hunk 1]
```diff
@@ -343,7 +343,10 @@ func (ep *endpoint) getNetworkFromStore() (*network, error) {
 		return nil, fmt.Errorf("invalid network object in endpoint %s", ep.Name())
 	}
 
-	return ep.network.ctrlr.getNetworkFromStore(ep.network.id)
+	ep.network.Lock()
```
Comment (aboch) [inline endpoint.go]: Can we reuse `network.getController()` ?
Ideally we could also do 
`ep.getNetwork().getController().getNetworkFromStore(ep.network.id)` 
but it looks too long of a concatenation...

**Comment by LK4D4** [comment]
@aboch changed, thanks!

**Comment by aboch** [comment]
LGTM

**Comment by mrjana** [comment]
LGTM

**Pull request merged**